src/samVisualize/commands/visualizeSamTemplate/samVisualization.ts

```diff
@@ -0,0 +1,227 @@
+                retainContextWhenHidden: true,
+            }
+        )
+        // During tests, a mock context is used, so the absolute paths will not resolve
+        try {
+            const graphObject = generateGraphFromYaml(textDocument.getText())
+
+            // Only generate a ResourceLineMap if the input has generated a valid GraphObject.
+            if (graphObject) {
+                this.resourceLineMap = generateResourceLineMap(textDocument.getText())
+            }
+
+            const graphObjectString = JSON.stringify(graphObject)
+
+            const primaryResourceListString = readFileSync(
+                this.extensionContext.asAbsolutePath(
+                    join('resources', 'light', 'samVisualize', 'primaryResources.json')
+                )
+            ).toString()
+
+            const iconsDir = this.extensionContext.asAbsolutePath(join('resources', 'light', 'samVisualize', 'icons'))
+
+            const iconsDataString = JSON.stringify(generateIconsMap(iconsDir, panel.webview))
+
+            const webviewStylesheetUri = panel.webview.asWebviewUri(
+                vscode.Uri.file(this.extensionContext.asAbsolutePath(join('media', 'css', 'samVisualize.css')))
+            )
+
+            const webviewBodyScriptUri = panel.webview.asWebviewUri(
+                // Change those file names
+                vscode.Uri.file(
+                    this.extensionContext.asAbsolutePath(join('samVisualizeRenderJS', 'samVisualizeRenderBundle.js'))
+                )
+            )
+            panel.webview.html = this.getWebviewContent(
+                graphObjectString,
+                primaryResourceListString,
+                iconsDataString,
+                webviewStylesheetUri,
+                webviewBodyScriptUri,
+                panel.webview.cspSource
+            )
+        } catch {}
+
+        // To close the visualization if the associated template no longer exists
+        this.disposables.push(
+            vscode.workspace.onDidCloseTextDocument(e => {
+                if (!this.templateDoesExist() && this.webviewPanel) {
+                    panel.dispose()
+                    vscode.window.showErrorMessage('Template associated with visualization closed or renamed')
+                }
+            })
+        )
+
+        const debounceInterval = 1000
+        const debouncedUpdate = _.debounce(this.updateVisualization.bind(this), debounceInterval)
+
+        // Update the visualization on template change
+        this.disposables.push(
+            vscode.workspace.onDidChangeTextDocument(async changedTextDocument => {
+                if (changedTextDocument.document.uri.path === this.textDocumentUri.path) {
+                    debouncedUpdate(changedTextDocument.document.getText())
+                }
+            })
+        )
+
+        const decorationType = vscode.window.createTextEditorDecorationType({
+            backgroundColor: 'rgb(135,206,250,0.3)', // Light sky blue
```

Comment:
What if you have a VS Code theme with that as the background color? I think you can set a value depending on whether or not the theme is considered dark.

Reply:
Good point, I'll make this more dynamic

Reply:
Update: I've changed it to a default selection, so there's no more colour involved. Same as selecting with the cursor.